src/lib/button-toggle/button-toggle.ts

```diff
@@ -278,14 +282,11 @@ export class MatButtonToggleGroup implements ControlValueAccessor, OnInit, After
       this._selectionModel.deselect(toggle);
     }
 
-    // Only emit the change event for user input.
-    if (isUserInput) {
-      this._emitChangeEvent();
+    if (deferEvents) {
```

Comment:
Add a comment here that explains why this is used? 

I'm generally not a fan of [flag arguments](https://martinfowler.com/bliki/FlagArgument.html), but I don't have another suggestion here that wouldn't involve more work.

Reply:
I'm not a fan of them either, but in this case it's a private API so it should be fine.

Reply:
Yeah, I'm okay with it going in. I think it would be useful to have a comment here like
```
// We defer emitting the change event in some cases because...
```

Reply:
Done.